src/libraries/System.Private.CoreLib/src/System/Activator.RuntimeType.cs

```diff
@@ -136,8 +136,20 @@ public static partial class Activator
 
         [Intrinsic]
         public static T CreateInstance<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] T>()
+            where T : allows ref struct
         {
-            return (T)((RuntimeType)typeof(T)).CreateInstanceOfT()!;
+            var rtType = (RuntimeType)typeof(T);
+            if (!rtType.IsValueType)
+            {
+                object o = rtType.CreateInstanceOfT()!;
+                return Unsafe.As<object, T>(ref o);
```

Comment:
Did you consider adding a comment specifying why this specifically is safe to use here? Left undocumented customers, internal or external, may think it's valid ot down cast arbitrary `object` to `ref struct`.

Reply:
I love that reasoning for a comment. Thank you.